jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt

```diff
@@ -261,16 +234,44 @@ class DotNetSamDebugSupport : SamDebugSupport {
         return promise
     }
 
-    private fun createNetCoreStartInfo(state: SamRunningState): DotNetCoreExeStartInfo =
-        DotNetCoreExeStartInfo(
-            dotNetCoreInfo = DotNetCoreInfo(REMOTE_NETCORE_CLI_PATH, null),
-            exePath = REMOTE_LAMBDA_COMPILED_PATH,
-            workingDirectory = "",
-            arguments = state.settings.handler,
-            environmentVariables = emptyList(),
-            runtimeArguments = null,
-            executeAsIs = false,
-            useExternalConsole = false,
-            needToBeInitializedImmediately = true
+    private suspend fun findDockerContainer(frontendPort: Int): String = spinUntilResult(Duration.ofSeconds(30)) {
+        ExecUtil.execAndGetOutput(
+            GeneralCommandLine(
+                "docker",
+                "ps",
+                "-q",
+                "-f",
+                "publish=${frontendPort}"
+            )
+        ).stdout.trim().nullize()
+    }
+
+    private suspend fun findDotnetPid(dockerContainer: String): Int = spinUntilResult(Duration.ofSeconds(30)) {
+        ExecUtil.execAndGetOutput(
+            GeneralCommandLine(
+                "docker",
+                "exec",
+                "-i",
+                dockerContainer,
+                "/usr/bin/pgrep",
```

Comment:
This won't work on windows?

Reply:
This is executed inside of the container which is Amazon Linux 2 based